nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -193,4 +205,76 @@ protected void notifyEvent(CoreSession session, String eventType, DocumentModel
         notifyEvent(session, eventType, commentParent, commentDocumentModel);
     }
 
+    /**
+     * Checks if a document has comments.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected abstract boolean isDocumentNotCommented(CoreSession session, DocumentModel document);
+
+    /**
+     * Checks if a document has comments from a particular user.
+     *
+     * @param session the core session
+     * @param document the document model who's comments are being counted
+     * @return true if comments were found
+     * @since 11.1
+     */
+    protected abstract boolean isDocumentNotCommentedByUser(CoreSession session, String author, DocumentModel document);
+
+    /**
+     * Subscribes top level document's author on first comment and commentor on his first comment if autosubscribtion is
+     * turned on.
+     *
+     * @param session the core session
+     * @param commentDocModel the comment
+     * @param document the document being commented
+     * @since 11.1
+     */
+    protected void autosubscribeToNewCommentsNotifications(CoreSession session, DocumentModel commentDocModel,
+            DocumentModel document) {
+        ConfigurationService configService = Framework.getService(ConfigurationService.class);
+
+        if (configService.isBooleanFalse(AUTOSUBSCRIBE_CONFIG_KEY)) {
+            return;
+        }
+
+        NuxeoPrincipal topLevelDocumentAuthor = getAuthor(document);
+        if (isDocumentNotCommented(session, document)) {
+            // Document author is subscribed on first comment by anybody
+            subscribeToNeCommentNotifications(document, topLevelDocumentAuthor);
+
+        }
+
+        NuxeoPrincipal commentAuthor = getAuthor(commentDocModel);
+        if (topLevelDocumentAuthor.getName().equals(commentAuthor.getName())) {
+            // Document author is comment author. He doesn't need to be resubscribed
+            return;
+        }
+
+        if (isDocumentNotCommentedByUser(session, commentAuthor.getName(), document)) {
+            // Comment author is writing his first comment on the document
+            subscribeToNeCommentNotifications(document, commentAuthor);
+        }
+    }
+
+    /**
+     * Subscribes a user to notifications on the document.
```

Comment:
State here that this method is subscribing the user to "commentAdded" notification, it does not subscribe the user to all notifications :)